stixcore/processing/pipeline.py

```diff
@@ -0,0 +1,72 @@
+import re
+import time
+import logging
+from pathlib import Path
+
+from watchdog.events import FileSystemEventHandler, LoggingEventHandler
+from watchdog.observers import Observer
+
+from stixcore.config.config import CONFIG
+from stixcore.io.soc.manager import SOCPacketFile
+from stixcore.processing.L0toL1 import Level1
+from stixcore.processing.LBtoL0 import Level0
+from stixcore.processing.TMTCtoLB import process_tmtc_to_levelbinary
+from stixcore.util.logging import get_logger
+
+logger = get_logger(__name__, level=logging.DEBUG)
+
+TM_REGEX = re.compile(r'.*PktTmRaw.*.xml$')
+
+
+class TMTCFileHandler(FileSystemEventHandler):
+    """
+    Handler to detect and process new TM files
+
+    As rsync is used to transfer the files to process need to take into account how rsync works.
+    Rsync works by first creating a temporary file of the same with name as the file being
+    transferred `myfile.xml` with an random extra extension `myfile.xml.NmRJ4x` it then
+    transfers the data to the temporary file and once the transfer is complete it them move/renames
+    the file back to the original name `myfile.xml`. Can detect file move event that match the TM
+    filename pattern.
+    """
+    def __init__(self, func):
+        """
+
+        Parameters
+        ----------
+        func : `callable`
+            The method to call when new TM is received
+        """
+        if not callable(func):
+            raise TypeError('func must be a callable')
+        self.func = func
+
+    def on_moved(self, event):
+        if TM_REGEX.match(event.dest_path):
+            self.func(Path(event.dest_path))
+
+
+def process(path):
+    lb_files = process_tmtc_to_levelbinary([SOCPacketFile(path)])
```

Comment:
Not sure if the list of file returned will be unique - maybe need to create set before they are returned here